visualization.py

```diff
@@ -0,0 +1,78 @@
+"""Visualization for Geometric Statistics."""
+
+# import pylab
+
+import numpy as np
+from matplotlib import pyplot as plt
+from mpl_toolkits.mplot3d import Axes3D
+
+# import rigid_transformations
+import rotations
+
+
+class Arrow3D():
+    "An arrow in 3d, i.e. a point and a vector."
+    def __init__(self, point, vector):
+        self.point = point
+        self.vector = vector
+
+    def draw(self, ax, **kwargs):
+        "Draw the arrow in 3D plot."
+        ax.quiver(self.point[0], self.point[1], self.point[2],
+                  self.vector[0], self.vector[1], self.vector[2],
+                  **kwargs)
+
+
+class Trihedron():
+    "A trihedron, i.e. 3 Arrow3Ds at the same point."
+    def __init__(self, point, vec_1, vec_2, vec_3):
+        self.arrow_1 = Arrow3D(point, vec_1)
+        self.arrow_2 = Arrow3D(point, vec_2)
+        self.arrow_3 = Arrow3D(point, vec_3)
+
+    def draw(self, ax, **kwargs):
+        """
+        Draw the trihedron by drawing its 3 Arrow3Ds.
+        Arrows are drawn is order using green, red, and blue
+        to show the trihedron's orientation.
+        """
+        self.arrow_1.draw(ax, color='g', **kwargs)
+        self.arrow_2.draw(ax, color='r', **kwargs)
+        self.arrow_3.draw(ax, color='b', **kwargs)
+
+
+def trihedron_from_rigid_transformation(transfo):
+    """
+    Transform a rigid transformation
+    into a trihedron s.t.:
+    - the trihedron's base point is the translation of the origin
+    of R^3 by the translation part of transfo,
+    - the trihedron's orientation is the rotation of the canonical basis
+    of R^3 by the rotation part of transfo.
+    """
+    translation = transfo[3:6]
+    rot_vec = transfo[0:3]
+    rot_mat = rotations.rotation_matrix_from_rotation_vector(rot_vec)
+    rot_mat = rotations.closest_rotation_matrix(rot_mat)
+
+    basis_vec_1 = np.array([1, 0, 0])
+    basis_vec_2 = np.array([0, 1, 0])
+    basis_vec_3 = np.array([0, 0, 1])
+
+    trihedron_vec_1 = np.dot(rot_mat, basis_vec_1)
+    trihedron_vec_2 = np.dot(rot_mat, basis_vec_2)
+    trihedron_vec_3 = np.dot(rot_mat, basis_vec_3)
+    trihedron = Trihedron(translation,
+                          trihedron_vec_1,
+                          trihedron_vec_2,
+                          trihedron_vec_3)
+
+    return trihedron
+
+
+def plot_trihedron(trihedron, **kwargs):
+    "Plot the trihedron."
+    fig = plt.figure()
+    ax = fig.add_subplot(111, projection='3d')
+    trihedron.draw(ax, **kwargs)
+    plt.show()
```

Comment:
you may also want to return a figure instead of showing here. And then ppl can use the figure how they see fit (like save it etc.)